java/clients/src/main/java/sleeper/clients/status/update/DeleteTable.java

```diff
@@ -59,9 +59,9 @@ public DeleteTable(InstanceProperties instanceProperties, AmazonS3 s3Client, Tab
 
     public void delete(String tableName) throws StateStoreException {
         TableProperties tableProperties = tablePropertiesStore.loadByName(tableName);
-        deleteAllObjectsInBucketWithPrefix(s3Client, instanceProperties.get(DATA_BUCKET), tableProperties.get(TABLE_ID));
         stateStoreProvider.getStateStore(tableProperties).clearSleeperTable();
         tablePropertiesStore.deleteByName(tableName);
+        deleteAllObjectsInBucketWithPrefix(s3Client, instanceProperties.get(DATA_BUCKET), tableProperties.get(TABLE_ID));
```

Comment:
When we call `tablePropertiesStore.deleteByName`, it will also delete the table index entry. It might be better to delete from the data bucket before we do that, so that if deleting from the bucket fails, we still have a record that that table existed.